src/main/java/az/keytd/expensetracker/service/UserService.java

```diff
@@ -32,30 +29,13 @@ public UserDetails loadUserByUsername(String firstName) throws UsernameNotFoundE
 
     }
 
-    // this method not needed, remove
-    public List<User> findByAllAddress(String address) {
-        return userRepository.findAllByAddress(address);
-    }
-
     public User getByEmail(String email) {
         User user = userRepository.findByEmail(email)
                 .orElseThrow(() -> new NotFoundException(email + " not found"));
-//you do not need save it you found above just return user;
-// or you just can write:
-// return  userRepository.findByEmail(email)
-// .orElseThrow(() -> new NotFoundException(email + " not found"));
-// for simplicity
-        return userRepository.save(user);
-    }
-
-    //TODO: here don't return optional, handle if user does not exist by email case in orElseThrow
-    // and you actually do not need this method you already have one getByEmail()
-    public Optional<User> findByEmail(String email) {
-        return userRepository.findByEmail(email);
+        return userRepository.getByEmail(email);
```

Comment:
here why not directly return user? why you calling repo again, you already have user